hfinger/hfinger_core.py

```diff
@@ -48,47 +47,43 @@ def get_hdr_case(hdr):
 
 # Checking method and proto version
 def get_method_version(request_split):
-    r_ver = ""
-    r_meth = ""
-    # Checking if HTTP version is provided, if not assuming it is HTTP 0.9 per www.w3.org/Protocols/HTTP/Request.html
+    req_version = ""
+    req_method = ""
+    # Checking if HTTP version is provided,
+    # if not assuming it is HTTP 0.9 per www.w3.org/Protocols/HTTP/Request.html
     if " HTTP/" not in request_split[0]:
-        r_ver = "9"
-        # take first seven characters of the first line of request to look for method (methods have up to 7 chars)
-        t2 = request_split[0][:7].upper().strip(" ")
-        # if method shorter than 7 chars we will have part of URL in t2
-        # we should find space between method and URL and cut the string on it
-        it = t2.find(" ")
-        meth = t2[:it]
-        if it < 0:
-            # method has 7 chars, so no need to cut t2
-            meth = t2
-        if meth in METHODS:
-            r_meth = meth[:2]
-    else:
-        t = request_split[0].split(" HTTP/")
-        t1 = t[0].lstrip(" ")
-        # check if method is present by taking first 7 characters and searching there for method
+        req_version = "9"
+        # take first seven characters of the first line of request to look for method
         # (methods have up to 7 chars)
-        t2 = t1[:7].upper().strip(" ")
-        # if method shorter than 7 chars we will have part of URL in t2
+        method_raw = request_split[0][:7].upper().strip(" ")
```

Comment:
this seem super convoluted, why not first split to get the method and then check its length?

Reply:
Updated in https://github.com/CERT-Polska/hfinger/pull/6/commits/7df77207ebb3ef8962961357c1e571c5dc55473e.